src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1946,8 +2005,17 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
 	block.size = static_cast<uint64_t>(cur_len);
 
 	if ((ret = blockDir->get(dpp, &block, y)) < 0) {
-	  ldpp_dout(dpp, 0) << "Failed to retrieve directory entry for: " << source->get_name() << " blockid: " << fst << " block size: " << cur_len << ", ret=" << ret << dendl;
-	  return ret;
+          if (ret == -ENOENT) {
```

Comment:
did we not decide that for non-dirty objects, the deletion will be taken care of by eviction. We only need to remove the block directory entries so that the object is not found in the cache anymore.